### PR TITLE
[WIP] experiment getTransaction from web3 (#1762)

### PR DESCRIPTION
**Pull request by tbenr**
This is a PR to show some experiments I did around #1762 

it currently:
- get current block and recursively search for 50 blocks any transaction related to the current account
- prints some stats on stdout. In my iOS simulator take 7.6 seconds to finish, Rinkeby testnet.

as suspected it is very slow. I don't think even a multithreaded solution gathering blocks in parallel can reach an acceptable performance to search for a meaningful time frame in the past (days?).

The hybrid solution to still get transactions from etherscan could be enhanced by adding a lookup on web3 to check them, but txs not returned by etherscan will never be found.

Hope it helps.
Any suggestion on if\how to procede is welcome!

### CODE REVIEW

**Comment by oskarth** [comment]
Sorry about no one getting back to you earlier @tbenr! Is it still WIP?

Requested review based on suggestions.

**Comment by jeluard** [comment]
Interesting experiment! I guess we don't want to merge it as is given the performance cost?
Could be something that we can activate based on user settings. This way users could choose not to rely on etherscan.

**Comment by tbenr** [comment]
@oskarth this is currently on-hold but my intention is to continue experimenting as suggested by @jeluard

**Comment by debragail** [comment]
still curious if there are any perftools to check performance for this?

**Comment by StatusSceptre** [comment]
Automation has closed this but it seems like a worthy venture still for a user option. Just seeing if experimentation still pursued here.

**Comment by Swader** [comment]
I would classify security / reliability of balance checking in the following order from least to most secure:

- Etherscan API
- RPC to a node (hardcoded)
- RPC to a node (user provided)
- LES
- Reading and verifying the chain

My suggestion to avoid the unsafe and centralized Etherscan is to add an option to connect Web3-style to a running, synced node. Status can provide a hardcoded whitelist (infura, Status' own node, I'll donate one at Bitfalls HQ and another at Audithor.io HQ) and users can select from those, but user should also be able to input their own if they're running a node. This is until we solve the light client problem.

In other words, let Status' web3 instance connect to custom providers.

**Comment by oskarth** [comment]
Closed PR as inactive. Please reopen if this is incorrect. (not you StatusSceptre, whoever you are)

**Comment by StatusSceptre** [comment]
@oskart see comment here 

https://github.com/status-im/status-react/issues/1762#issuecomment-458851543

Will let @tbenr reopen when he’s got more work done on it.
